Area51/Sources/FeedViewController.swift

```diff
@@ -1,66 +1,74 @@
+import UIKit
 import ListingService
 import SafariServices
-import UIKit
 
 final class FeedViewController: UIViewController {
-    @IBOutlet weak var tableView: UITableView!
-    private var dataSource: ListingsDataSource! {
-        didSet {
-            self.dataSource.updated = { [weak self] in
-                self?.tableView.refreshControl?.endRefreshing()
-                self?.tableView.reloadData()
-            }
-        }
-    }
+    @IBOutlet private weak var tableView: UITableView!
+    private var viewModel: FeedViewModel!
 
     class func instantiateFromStoryboard(withSubreddit subreddit: Subreddit) -> FeedViewController {
         let storyboard = UIStoryboard.feed
         let feedViewController: FeedViewController = storyboard.viewControllerFrom(identifier: "FeedViewController")
-        feedViewController.dataSource = ListingsDataSource(subreddit: subreddit)
-        feedViewController.title = subreddit.name
+        feedViewController.viewModel = FeedViewModel(with: subreddit)
         return feedViewController
     }
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.setupTableView()
+        setup()
+    }
+}
+
+private extension FeedViewController {
+    func setup() {
+        title = viewModel.subredditName
+        setupTableView()
+        bindViewModel()
     }
 
-    private func setupTableView() {
-        self.tableView.delegate = self
-        self.tableView.dataSource = self
```

Comment:
When I run the app, feed does not load. Guessing because these are missing.

Reply:
Crap. Sorry about that. I actually set that in the storyboard since we defined the UITableView there - less code that way. Updated the code.